testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -853,6 +854,97 @@ async fn test_delayed_decryption_latest_event() -> Result<()> {
+    alice_sync_service.start().await;
+
+    let bob_sync_service = SyncService::builder(bob.clone()).build().await.unwrap();
+    bob_sync_service.start().await;
+
+    // alice creates a room and invites bob.
+    let alice_room = alice
+        .create_room(assign!(CreateRoomRequest::new(), {
+            invite: vec![],
+            is_direct: true,
+            preset: Some(RoomPreset::PrivateChat),
+        }))
+        .await?;
+    alice_room.enable_encryption().await?;
+
+    // Initial message to make sure any lazy /members call is performed before the
+    // test actually starts
+    alice_room
+        .send(RoomMessageEventContent::text_plain("dummy message to make members call"))
+        .await?;
+
+    // Set up sliding sync for bob.
+    let sliding_bob = bob
+        .sliding_sync("main")?
+        .with_all_extensions()
+        .poll_timeout(Duration::from_secs(2))
+        .network_timeout(Duration::from_secs(2))
+        .add_list(
+            SlidingSyncList::builder("all")
+                .sync_mode(SlidingSyncMode::new_selective().add_range(0..=20)),
+        )
+        .build()
+        .await?;
+
+    let s = sliding_bob.clone();
+    spawn(async move {
+        let stream = s.sync();
+        pin_mut!(stream);
+        while let Some(up) = stream.next().await {
+            warn!("bob received update: {up:?}");
+        }
+    });
+
+    // Send the invite to Bob and a message to reproduce the edge case
+    alice_room.invite_user_by_id(bob.user_id().unwrap()).await.unwrap();
+    alice_room.send(RoomMessageEventContent::text_plain("hello world")).await?;
+
+    // Join the room from Bob's client
+    let bob_room = bob.get_room(alice_room.room_id()).unwrap();
+    bob_room.join().await?;
+
+    assert_eq!(alice_room.state(), RoomState::Joined);
+    assert!(alice_room.is_encrypted().await.unwrap());
+    assert_eq!(bob_room.state(), RoomState::Joined);
+    assert!(bob_room.is_encrypted().await.unwrap());
+
+    let bob_timeline = bob_room.timeline_builder().build().await?;
+    let (_, timeline_stream) = bob_timeline.subscribe().await;
+    pin_mut!(timeline_stream);
+
+    // Get previous events, including the sent message
+    bob_timeline.paginate_backwards(3).await?;
+
+    // Look for the sent message, which should not be an UTD event
+    loop {
+        let diff = timeout(Duration::from_millis(100), timeline_stream.next())
+            .await
+            .expect("Timed out. Neither an UTD or the sent message were found")
+            .unwrap();
+        if let VectorDiff::PushFront { value } = diff {
+            if let Some(event) = value.as_event() {
+                if let Some(message) = event.content().as_message() {
+                    if message.body() == "hello world" {
+                        return Ok(());
+                    } else {
+                        panic!("Unexpected message event found");
+                    }
```

Comment:
nit: else after return.

```suggestion
                    if message.body() == "hello world" {
                        return Ok(());
                    }
                    panic!("Unexpected message event found");
```

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3672/commits/75dfe35c127d4f4bb768126a4adef21878649c4a